lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,528 @@
+	targetUrl := getPresignURL()
+
+	r, err := http.NewRequest(http.MethodGet, targetUrl.String(), nil)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	r.Header.Add(awsHeaderClusterName, clusterName)
+	creds, err := credsProvider.Retrieve(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	signer := v4.NewSigner()
+	presigned, _, err := signer.PresignHTTP(ctx, creds, r, hashForGetRequests, "sts", region, timestamp)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	kubeToken := awsKubePrefix + base64.RawURLEncoding.EncodeToString([]byte(presigned))
+
+	eksClusterCA, err := base64.StdEncoding.DecodeString(clusterCA)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	restConfig := &rest.Config{
+		Host:        clusterEndpoint,
+		BearerToken: kubeToken,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: eksClusterCA,
+		},
+	}
+
+	configFlags := genericclioptions.NewConfigFlags(false)
+	configFlags.WithWrapConfigFn(func(*rest.Config) *rest.Config {
+		return restConfig
+	})
+
+	return configFlags, nil
+}
+
+func getHelmActionConfig(clientGetter genericclioptions.RESTClientGetter, log logrus.FieldLogger) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(clientGetter, agentNamespace, "secret", log.WithField("source", "helm").Debugf); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return actionConfig, nil
+}
+
+func getHelmSettings() *helmCli.EnvSettings {
+	helmSettings := helmCli.New()
+	helmSettings.RepositoryCache = os.TempDir()
+	helmSettings.SetNamespace(agentNamespace)
+
+	return helmSettings
+}
+
+// checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
+func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
+	listCmd := action.NewList(actionConfig)
+
+	releases, err := listCmd.Run()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	for _, release := range releases {
+		if release.Name == agentName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// installKubeAgent installs teleport-kube-agent chart to the target EKS cluster.
+func installKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr, joinToken, resourceId string, actionConfig *action.Configuration, settings *helmCli.EnvSettings, req EnrollEKSClustersRequest) error {
+	installCmd := action.NewInstall(actionConfig)
+	installCmd.RepoURL = agentRepoURL
+	installCmd.Version = req.AgentVersion
+	if strings.Contains(installCmd.Version, "dev") {
+		installCmd.Version = "" // For testing during development.
+	}
+
+	chartPath, err := installCmd.LocateChart(agentName, settings)
+	if err != nil {
+		return trace.Wrap(err, "could not locate chart")
+	}
+	defer func() {
+		// Clean up downloaded chart.
+		_ = os.Remove(chartPath)
```

Comment:
```suggestion
		if err = os.RemoveAll(chartPath); err!=nil{
		    log....
		
		}
```

`os.Remove` will fail because the dir isn't empty
>Remove removes the named file or (empty) directory. If there is an error, it will be of type *PathError.

Reply:
chartPath is a single gz file though.

Reply:
https://github.com/gravitational/teleport/pull/36594/commits/4b8cb56620b2de458ce588122c31cdc1fa803918